public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -1,85 +1,55 @@
 
-export class PromExploreQueryEditor extends PureComponent<Props, State> {
-  // Query target to be modified and used for queries
-  query: PromQuery;
+export function PromExploreQueryEditor(props: Props) {
+  const { query, data, datasource, history, onChange, onRunQuery } = props;
 
-  constructor(props: Props) {
-    super(props);
-    const { query } = props;
-    this.query = query;
-    // Query target properties that are fully controlled inputs
-    this.state = {
-      // Fully controlled text inputs
-      interval: query.interval,
-    };
+  function onChangeQueryStep(value: string) {
+    const { query, onChange } = props;
+    const nextQuery = { ...query, interval: value };
+    onChange(nextQuery);
   }
 
-  onFieldChange = (query: PromQuery, override?: any) => {
-    this.query.expr = query.expr;
-  };
-
-  onIntervalChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
-    const interval = e.currentTarget.value;
-    this.query.interval = interval;
-    this.setState({ interval });
-  };
-
-  onRunQuery = () => {
-    const { query } = this;
-    this.props.onChange(query);
-    this.props.onRunQuery();
-  };
+  function onStepChange(e: React.SyntheticEvent<HTMLInputElement>) {
+    if (e.currentTarget.value !== query.interval) {
+      onChangeQueryStep(e.currentTarget.value);
+    }
+  }
 
-  onReturnKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+  function onReturnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
     if (e.key === 'Enter') {
-      this.onRunQuery();
+      onRunQuery();
     }
-  };
-
-  render() {
-    const { datasource, query, data, history } = this.props;
-    const { interval } = this.state;
-
-    return (
-      <div className="gf-form-inline">
-        <PromQueryField
-          datasource={datasource}
-          query={query}
-          onRunQuery={this.onRunQuery}
-          onChange={this.onFieldChange}
-          history={history}
-          data={data}
-        >
-          <div className="gf-form-inline explore-input--ml">
-            <div className="gf-form">
-              <FormLabel width={4}>Step</FormLabel>
-              <input
-                type="text"
-                className="gf-form-input width-6"
-                placeholder={'auto'}
-                onChange={this.onIntervalChange}
-                onKeyDown={this.onReturnKeyDown}
-                value={interval}
-              />
-            </div>
-          </div>
-        </PromQueryField>
-      </div>
-    );
   }
+
+  return (
+    <PromQueryField
+      datasource={datasource}
+      query={query}
+      onRunQuery={onRunQuery}
+      onChange={onChange}
+      history={history}
+      data={data}
+      ExtraFieldElement={
+        <PromExploreExtraField
+          label={'Step'}
+          onChangeFunc={onStepChange}
+          onKeyDownFunc={onReturnKeyDown}
+          value={query.interval}
+        />
+      }
+    />
+  );
 }
+
+export default memo(PromExploreQueryEditor);
```

Comment:
Is there need to have both memo and non memo exports? If previously something used the `PromExploreQueryEditor` it got the PureComponent version. Now we changed that to non memoized version and add a new export for the memoized one but that seems to be used only in test.

Reply:
Well, we use memoization to get some slight performance improvements (so it's going to be used as a default way to use a given component), but the non-memoized version is useful for testing as enzyme still has problems working with memos.

Reply:
But if I see correctly the default memoized version is used only in the test, while the non memoized version is going to be used everywhere else as that is what module.ts uses, or am I seeing that wrong?

Reply:
Memoized export should be used pretty much everywhere, non-memoized one is only for tests. Fixed the import in `module.ts`.